portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -69,25 +154,27 @@ private String convertToJson(UserAuth userAuth) {
     private String loginUrl;
     private String logoutUrl;
     private String email;
+    private String username;
 
     // Constructor to create UserAuth object with no user logged in.
-    // Empty strings represent no value (null is avoided).
+    // Empty strings represent no value (null is avoided to prevent errors).
     private UserAuth(String loginUrl) {
-      this(false, loginUrl, "", "");
+      this(false, loginUrl, "", "", "");
```

Comment:
I think Bootsie mentioned this in an earlier PR but generally 'null' is a better option when you want to indicate 'there's nothing meaningful here'. Of course, this comes with a cost of handling the possibility that these fields are null elsewhere in your code.

Reply:
Ok, I'll use `null` -- you're right that my early intuition was to use `""` just to prevent `NullPointerException` errors, but `null` fits better in this context.